NBISweden/urdr
# Reuse toast component to display URL warning

## Related issue(s) and PR(s)
This PR closes #444 and relates to #439.



## Type of change

 
## List of changes made

- the Toast component was made reusable. It takes the type (warning, info, success), a timeout and a message as arguments. The styling is applied depending on the type.
- The svg for the close button was moved from a separate file into a component so that we can use the `currentColor` property and change the color of the image depending on the type of toast.
- the toast got a little slide-in animation for seeing users to catch more attention
- because there now can be use cases where two toasts should be shown at the same time, that was made possible.
- the warning about the user typing a wrong URL was removed from the top of the page and is now displayed in a toast.
- the logic for when to display a warning was refactored using React's functionality of `useEffect` which makes the code shorter.

## Screenshot of the fix


## Testing

- open the Report page and change the URL to contain an invalid year (e.g. 20w20)
- you should see the report page with the current year / week displayed and the warning toast in the bottom left
- try the same with a valid year but invalid week, you should get the same result
- to make sure the "All changes saved!" toast still works, make a change and save. 
- You should see the success toast appear in the bottom left.

## Further comments


## Definition of Done checklist
